src/scripts/search.js

```diff
@@ -0,0 +1,135 @@
+import * as pagefind from '/pagefind/pagefind.js';
+
+const url = new URLSearchParams(window.location.search);
+
+pagefind.options({
+  baseUrl: '/',
+});
+
+const searchTerm = url.get('q');
+let index = url.get('page');
+let lang = 'en';
+
+if (window.location.href.includes('/fr/')) {
+  lang = 'fr';
+}
+
+if (!index) {
+  index = 1;
+}
+
+if (searchTerm) {
+  // I realized we don't have a value property for gcds-search, want to add that
+  // document.getElementById('searchbar').value = searchTerm;
+  const search = await pagefind.search(searchTerm);
+  const results = search.results;
+
+  const langObject = {
+    en: {
+      results: `${results.length} search results for "${searchTerm}"`,
+      noresults: 'No results',
+      noresultsbody: `
+        <p class="mb-400">No pages were found that match your search terms.</p>
+
+        <p class="mb-400">Suggestions:</p>
+
+        <ul class="list-disc mb-400">
+          <li>Make sure all search terms are spelled correctly</li>
+          <li>Try different search terms</li>
+          <li>Try more general search terms</li>
+        </ul>
+      `,
+    },
+      
+    fr: {
+      results: `${results.length} résultats de recherche pour « ${searchTerm} »`,
+      noresults: 'Aucun résultat',
+      noresultsbody: `
+        <p class="mb-400">Aucun résultat ne correspond à vos critères de recherche.</p>
+
+        <p class="mb-400">Suggestions:</p>
+
+        <ul class="list-disc mb-400">
+          <li>Assurez-vous que tous vos termes de recherches sont bien orthographiés</li>
+          <li>Utilisez de différents termes de recherche</li>
+          <li>Utilisez des termes de recherche plus généraux</li>
+        </ul>
+      `,
+    }
+  }
+
+  // Results length
+  if (results.length > 0) {
+    // Results heading
+    let resultsHeading = document.createElement('gcds-heading');
+    resultsHeading.setAttribute('tag', 'h2');
+    resultsHeading.innerHTML = langObject[lang].results;
+    document.getElementById('results-count').append(resultsHeading);
+
+    const totalPages = Math.ceil(results.length / 10);
+
+    let pageResults = results.slice(10 * (index - 1), 10 * index + 1);
```

Comment:
Can you add a variable for the number of results per page (aka `10`) to make it easier for potential future updates and better readability?